pcicrawler/ocp_diag.py

```diff
@@ -0,0 +1,259 @@
+                                self._test_name,
+                                self._device))
+            return True
+
+
+class CheckLocation(OCPTestStep):
+    _test_name = "pci_location_check"
+
+    def run(self):
+        return super().run(self._device.location)
+
+
+class CheckVendorID(OCPTestStep):
+    _test_name = "pci_vendor_id_check"
+
+    def run(self):
+        return super().run(self._device.vendor_id)
+
+
+class CheckDeviceID(OCPTestStep):
+    _test_name = "pci_device_id_check"
+
+    def run(self):
+        return super().run(self._device.device_id)
+
+
+class CheckClassID(OCPTestStep):
+    _test_name = "pci_class_id_check"
+
+    def run(self):
+        return super().run(self._device.class_id)
+
+
+class CheckPhysicalSlot(OCPTestStep):
+    _test_name = "pci_physical_slot_check"
+
+    def run(self):
+        return super().run(self._device.express_slot.slot)
+
+
+class CheckCurrentLinkSpeed(OCPTestStep):
+    _test_name = "pci_current_link_speed_check"
+
+    def run(self):
+        return super().run(self._device.express_link.cur_speed)
+
+
+class CheckCurrentLinkWidth(OCPTestStep):
+    _test_name = "pci_current_link_width_check"
+
+    def run(self):
+        return super().run(self._device.express_link.cur_width)
+
+
+class CheckCapableLinkSpeed(OCPTestStep):
+    _test_name = "pci_capable_link_speed_check"
+    
+    def run(self):
+        return super().run(self._device.express_link.capable_speed)
+
+
+class CheckCapableLinkWidth(OCPTestStep):
+    _test_name = "pci_capable_link_width_check"
+    
+    def run(self):
+        return super().run(self._device.express_link.capable_width)
+
+
+class CheckAddress(OCPTestStep):
+    _test_name = "pci_address_check"
+
+    def run(self):
+        dut = tv.Dut(id=str(self._device), name="PCIe Device")
+        with self._ocp_run.scope(dut=dut):
+            step = self._ocp_run.add_step(self._test_name)
+            if str(self._device) not in str(self._expected_result):
+                step.add_diagnosis(tv.DiagnosisType.FAIL, 
+                                verdict="Test {} did not find {} in expected list {}.".format(
+                                    self._test_name,
+                                    self._device,
+                                    self._expected_result))
+            else:
+                step.add_diagnosis(tv.DiagnosisType.PASS, 
+                                verdict="Test {} for device {} PASSED".format(
+                                    self._test_name,
+                                    self._device))
+
+
+class CheckAER(OCPTestStep):
+    _test_name = "pci_AER_check"
+
+    def __init__(self,
+                    device: PCIDevice,
+                    expected_results: str,
+                    ocp_run: tv.TestRun):\
+        super().__init__(device, expected_results, ocp_run)
+
+    def run(self):
+        return super().run(self._device.express_aer)
+
+
+class TestSet(object):
```

Comment:
missed  an inherit from object; also this could be a dataclass, it's just a collection of fields without other logic